client/ayon_core/tools/utils/thumbnail_paint_widget.py

```diff
@@ -33,19 +47,136 @@ def __init__(self, parent):
         self._use_checker = True
         self._checker_color_1 = QtGui.QColor(89, 89, 89)
         self._checker_color_2 = QtGui.QColor(188, 187, 187)
 
+        # Enable mouse tracking and set focus policy to receive mouse events
+        self.setMouseTracking(True)
+        self.setFocusPolicy(QtCore.Qt.ClickFocus)
+        # Ensure the widget can receive mouse events
+        self.setAttribute(QtCore.Qt.WA_MouseTracking, True)
+        self.setAttribute(QtCore.Qt.WA_AcceptTouchEvents, False)
+
+        self._setup_video_player()
+
+    def _setup_video_player(self):
+        """Setup video player components."""
+        if not HAS_QTMULTIMEDIA:
+            return
+
+        try:
+            # Qt6 API
+            self._media_player = QtMultimedia.QMediaPlayer(self)
+            self._video_widget = QtMultimediaWidgets.QVideoWidget(self)
+            self._video_widget.hide()
+            self._media_player.setVideoOutput(self._video_widget)
+
+            # Loop video playback - Qt6 uses different signal
+            self._media_player.mediaStatusChanged.connect(
+                self._on_media_status_changed
+            )
+        except (AttributeError, TypeError):
+            # Qt5 fallback
+            try:
+                self._media_player = QtMultimedia.QMediaPlayer(
+                    self, QtMultimedia.QMediaPlayer.VideoSurface
+                )
+                self._video_widget = QtMultimediaWidgets.QVideoWidget(self)
+                self._video_widget.hide()
+                self._media_player.setVideoOutput(self._video_widget)
+
+                # Loop video playback
+                self._media_player.mediaStatusChanged.connect(
+                    self._on_media_status_changed
+                )
+            except Exception as e:
+                print(f"Failed to setup video player: {e}")
+                self._media_player = None
+                self._video_widget = None
+
+    def _on_media_status_changed(self, status):
+        """Handle media status changes for looping."""
+        if not self._media_player:
+            return
+
+        # Qt6 uses QMediaPlayer.MediaStatus enum
+        try:
+            end_of_media = QtMultimedia.QMediaPlayer.MediaStatus.EndOfMedia
+        except AttributeError:
+            # Qt5 uses QMediaPlayer.EndOfMedia
+            end_of_media = QtMultimedia.QMediaPlayer.EndOfMedia
+
+        if status == end_of_media:
+            self._media_player.setPosition(0)
+            self._media_player.play()
+
+    def _is_video_file(self, filepath):
+        """Check if file is a video file (h264_*, mp4, mov, etc.)."""
+        if not filepath:
+            return False
+        basename = os.path.basename(filepath).lower()
+        # Check for h264_* naming convention
+        if "h264_" in basename or "h264" in basename:
+            return True
+        # Check for common video extensions
+        video_exts = (".mp4", ".mov", ".avi", ".mkv", ".webm", ".m4v")
```

Comment:
Should be constant and can be set.
```suggestion
        video_exts = {".mp4", ".mov", ".avi", ".mkv", ".webm", ".m4v"}
```